### PR TITLE
feat: downloading attachements when using remote-settings CLI

### PR DESCRIPTION
**Pull request by gruberb**
When using the CLI commands:

`cargo remote-settings dump-sync` and `cargo remote-settings dump-get COLLECTION_NAME`

we previously didn't download the attachments for each record. Now, we the two commands download automatically all attachments for each record if present. 

This PR also removes the `--create-pr` flag for the `cargo remote-settings dump-sync` command.

### CODE REVIEW

File: examples/remote-settings-cli/src/dump/client.rs [hunk 1]
```diff
@@ -223,7 +228,250 @@ impl CollectionDownloader {
...
+        let base_path = self
+            .output_dir
+            .join(DUMPS_DIR)
+            .join(bucket)
+            .join("attachments")
+            .join(collection);
+
+        (
+            base_path.join(format!("{}.bin", record_id)),
+            base_path.join(format!("{}.meta", record_id)),
```
Comment (leplatrem) [inline examples/remote-settings-cli/src/dump/client.rs:298]: Same here, it should be `.meta.json`

**Review by leplatrem — APPROVED** [review]
Great!

The file names must be adjusted for the bundles extraction to work as expected (and also to be aligned with desktop, see https://searchfox.org/mozilla-central/source/services/settings/dumps/main/search-config-icons)

Approving the PR to avoid blocking you because that's the only thing I could spot 😉

File: examples/remote-settings-cli/src/dump/client.rs [hunk 2]
```diff
@@ -223,7 +228,250 @@ impl CollectionDownloader {
...
+    ) -> (PathBuf, PathBuf) {
+        let base_path = self
+            .output_dir
+            .join(DUMPS_DIR)
+            .join(bucket)
+            .join("attachments")
+            .join(collection);
+
+        (
+            base_path.join(format!("{}.bin", record_id)),
```
[the remark this answers is not in the document]
Comment (gruberb) [reply examples/remote-settings-cli/src/dump/client.rs:297]: Thanks! I changed `.meta` to `.meta.json`. And for the "bin", I just remove the extension all together?

For example, for the `regions` collection, we have `world` and `world.meta.json` now.

[the remark this answers is not in the document]
Comment (leplatrem) [reply examples/remote-settings-cli/src/dump/client.rs]: Yep

**Pull request merged**